File: campuspool_server/server/src/main/java/com/campuspoolspring/model/ReservationStatus.java
package com.campuspoolspring.model; // 본인의 패키지 경로에 맞게 수정

public enum ReservationStatus {
    REQUESTED,  // 요청됨 (탑승자가 예약 요청)
    ACCEPTED,   // 수락됨 (운전자가 수락)
    // CONFIRMED, // 확정됨 (ACCEPTED와 동일하게 사용하거나, 별도 확정 단계가 있다면 추가)
    REJECTED,   // 거절됨 (운전자가 거절)
    CANCELLED   // 취소됨 (탑승자 또는 운전자가 취소)
}

File: campuspool_app/lib/models/reservation_status.dart
// lib/models/reservation_status.dart

enum ReservationStatus {
  REQUESTED,  // 요청됨
  ACCEPTED,   // 수락됨
  REJECTED,   // 거절됨
  CANCELLED,  // 취소됨
  NONE,       // 관련 예약 없음 (클라이언트에서 사용하기 위한 상태 또는 서버에서 명시적으로 줄 경우)
  UNKNOWN     // 파싱 실패 또는 알 수 없는 상태
}

ReservationStatus reservationStatusFromString(String? statusString) {
  if (statusString == null || statusString.isEmpty) {
    return ReservationStatus.NONE; // 또는 ReservationStatus.UNKNOWN 등 기본값
  }
  switch (statusString.toUpperCase()) {
    case 'REQUESTED':
      return ReservationStatus.REQUESTED;
    case 'ACCEPTED':
      return ReservationStatus.ACCEPTED;
    case 'REJECTED':
      return ReservationStatus.REJECTED;
    case 'CANCELLED':
      return ReservationStatus.CANCELLED;
    case 'NONE': // 서버에서 "NONE"으로 명시적으로 줄 경우
      return ReservationStatus.NONE;
    default:
      print('Unknown reservation status string: $statusString');
      return ReservationStatus.UNKNOWN;
  }
}